VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/judge-waiting-room/judge-waiting-room.component.ts

```diff
@@ -401,6 +402,7 @@ export class JudgeWaitingRoomComponent extends WaitingRoomBaseDirective implemen
 
     async joinHearingInSession() {
         await this.videoCallService.joinHearingInSession(this.conferenceId, this.participant.id);
+        this.shouldUpdateHostShowVideo = true;
```

Comment:
Should this go above the previous line?

Reply:
why you think should go above?

Reply:
> Does joinHearingInSession() or startHearing() eventually cause shouldCurrentUserJoinHearing() in the base component where this is used to get called? Just wondering if there's a potential race condition.

we don't do any change of shouldUpdateHostShowVideo in startHearing(), currently don't find there is any race condition, shouldUpdateHostShowVideo is doing to ask base component update video for the host who is starting or resume the hearing inmiadiately rather than any other hosts

Reply:
Sorry, I meant this.videoCallService.startHearing(). 

Isn't the idea that these functions call the api which does whatever it does and then at some stage the client receives an event via signalR that calls updateShowHearing() in the base component which checks this property?

The chances of the race condition being a problem are pretty much zero but I would think we should set the property before calling the api anyway.